scripts/create_cloud_image.py

```diff
@@ -70,25 +90,47 @@ def main():
         if image.name == args.imagename
     ]
     current_ids = [image.id for image in current_images]
+    print("Existing image IDs: {}".format(current_ids))
 
     conn.compute.create_server_image(
         server=args.serveruuid, name=args.imagename
     )
-
-    for image in conn.image.images():
-        if image.name == args.imagename and image.id not in current_ids:
-            saved_image = image
+    for _ in xrange(240):
+        try:
+            saved_image = find_new_image(conn, current_ids, args.imagename)
             break
-
-    timeout = 900
-    while timeout > 0 and saved_image.status not in ("active", "error"):
-        timeout -= 60
-        sleep(60)
+        except Exception:
+            sleep(60)
+    else:
+        raise Exception("Failed to get image after creating it: {i}"
+                        .format(i=args.imagename))
+
+    print("Waiting for image:{i} to achieve active status."
+          .format(i=saved_image.name))
+    for attempt in xrange(240):
```

Comment:
waiting 4 hours for an image to become active seems a bit excessive? Especially if we are waiting for this before failing the job.

Reply:
Given that the it's a large image, and not cached, sometimes it takes quite a bit of time to deploy. Four hours may be a bit much, but let's see how it goes.